### PR TITLE
x11-plugins/{wmmenu,wmmemload,wmmisc,wmmon+smp,wmnd}: EAPI7, improve ebuild

### PR DESCRIPTION
**Pull request by mm1ke**
Hi,

Another round of fixes. Mostly simple changes.
Please review.

diff -u wmmenu:
```
--- wmmenu-1.3-r1.ebuild        2018-05-06 12:15:30.746161257 +0200
+++ wmmenu-1.3-r2.ebuild        2018-07-22 19:26:53.232963668 +0200
@@ -1,8 +1,8 @@
 # Copyright 1999-2018 Gentoo Foundation
 # Distributed under the terms of the GNU General Public License v2
 
-EAPI=5
-inherit eutils toolchain-funcs
+EAPI=7
+inherit toolchain-funcs
 
 DESCRIPTION="a popup menu of icons like in AfterStep, as a dockapp"
 HOMEPAGE="https://www.dockapps.net/wmmenu"
@@ -10,18 +10,20 @@
 
 LICENSE="GPL-2"
 SLOT="0"
-KEYWORDS="amd64 x86"
-IUSE=""
+KEYWORDS="~amd64 ~x86"
 
 RDEPEND="x11-libs/gdk-pixbuf[X]
        >=x11-libs/libdockapp-0.7:="
 DEPEND="${RDEPEND}"
 
-S=${WORKDIR}/dockapps
+S="${WORKDIR}/dockapps"
 
-src_prepare() {
-       epatch "${FILESDIR}"/${P}-Makefile.patch
+PATCHES=( "${FILESDIR}"/${P}-Makefile.patch )
+
+DOCS=( README TODO example/{apps,defaults,extract_icon_back} )
 
+src_prepare() {
+       default
        sed -e 's#<dockapp.h>#<libdockapp/dockapp.h>#' -i *.c || die
 }
 
@@ -32,5 +34,5 @@
 src_install() {
        dobin wmmenu
        doman wmmenu.1
-       dodoc README TODO example/apps example/defaults example/extract_icon_back
+       einstalldocs
 }
```

diff -u wmmemload:
```--- wmmemload-0.1.8.ebuild      2018-05-23 19:05:41.675212458 +0200
+++ wmmemload-0.1.8-r1.ebuild   2018-07-22 19:29:18.700831338 +0200
@@ -1,8 +1,8 @@
 # Copyright 1999-2018 Gentoo Foundation
 # Distributed under the terms of the GNU General Public License v2
 
-EAPI=5
-inherit autotools eutils
+EAPI=7
+inherit autotools
 
 DESCRIPTION="dockapp that displays memory and swap space usage"
 HOMEPAGE="https://www.dockapps.net/wmmemload"
@@ -10,8 +10,7 @@
 
 LICENSE="GPL-2"
 SLOT="0"
-KEYWORDS="alpha amd64 ~ppc ppc64 sparc x86"
-IUSE=""
+KEYWORDS="~alpha ~amd64 ~ppc ~ppc64 ~sparc ~x86"
 
 RDEPEND="x11-libs/libX11
        x11-libs/libXext
@@ -21,13 +20,9 @@
        x11-libs/libICE
        x11-libs/libXt"
 
-S=${WORKDIR}/dockapps
+S="${WORKDIR}/dockapps"
 
 src_prepare() {
+       default
        eautoreconf
 }
-
-src_install() {
-       emake DESTDIR="${D}" install
-       dodoc AUTHORS ChangeLog THANKS
-}

```

diff -u wmmisc:
```
--- wmmisc-1.1.ebuild   2018-05-23 19:05:41.675212458 +0200
+++ wmmisc-1.1-r1.ebuild        2018-07-22 19:34:34.633284226 +0200
@@ -1,9 +1,9 @@
 # Copyright 1999-2018 Gentoo Foundation
 # Distributed under the terms of the GNU General Public License v2
 
-EAPI=0
+EAPI=7
 
-inherit eutils toolchain-funcs
+inherit toolchain-funcs
 
 DESCRIPTION="a monitoring dockapp for system load, user amount, fork amount and processes"
 HOMEPAGE="https://packages.qa.debian.org/w/wmmisc.html"
@@ -11,10 +11,9 @@
 
 LICENSE="GPL-2"
 SLOT="0"
-KEYWORDS="~amd64 ~ppc x86"
-IUSE=""
+KEYWORDS="~amd64 ~ppc ~x86"
 
-S=${WORKDIR}/${P}/src
+S="${WORKDIR}/${P}/src"
 
 RDEPEND="x11-libs/libX11
        x11-libs/libXext
@@ -22,15 +21,11 @@
 DEPEND="${RDEPEND}
        x11-base/xorg-proto"
 
-src_unpack() {
-       unpack ${A}
-       cd "${S}"
-       epatch "${FILESDIR}"/${P}-build.patch
-}
+PATCHES=( "${FILESDIR}"/${P}-build.patch )
 
 src_compile() {
        tc-export CC
-       emake || die "emake failed."
+       emake
 }
 
 src_install() {
```

diff -u wmmon+smp:
```
--- wmmon+smp-1.0-r2.ebuild     2018-05-23 19:05:41.675212458 +0200
+++ wmmon+smp-1.0-r3.ebuild     2018-07-22 19:42:23.913255661 +0200
@@ -1,18 +1,17 @@
 # Copyright 1999-2018 Gentoo Foundation
 # Distributed under the terms of the GNU General Public License v2
 
-EAPI=5
+EAPI=7
 
-inherit eutils toolchain-funcs
+inherit toolchain-funcs
 
 DESCRIPTION="Dockapp CPU monitor resembling Xosview, support for smp"
-SRC_URI="http://www.ne.jp/asahi/linux/timecop/software/wmmon+smp.tar.gz"
 HOMEPAGE="http://www.ne.jp/asahi/linux/timecop/"
+SRC_URI="http://www.ne.jp/asahi/linux/timecop/software/wmmon+smp.tar.gz"
 
 SLOT="0"
 LICENSE="GPL-2"
-KEYWORDS="amd64 ppc64 x86"
-IUSE=""
+KEYWORDS="~amd64 ~ppc64 ~x86"
 
 RDEPEND="
        x11-libs/libX11
@@ -21,9 +20,10 @@
 DEPEND="${RDEPEND}
        x11-base/xorg-proto"
 
-S=${WORKDIR}/wmmon.app/wmmon
+S="${WORKDIR}/wmmon.app/wmmon"
 
 src_prepare() {
+       default
        # Respect LDFLAGS, see bug #335047
        sed \
                -e 's/cc -o/${CC} ${LDFLAGS} -o/' \
@@ -32,8 +32,7 @@
        tc-export CC
 
        cd "${WORKDIR}"/wmmon.app || die
-       ls
-       epatch "${FILESDIR}"/${P}-list.patch
+       eapply "${FILESDIR}"/${P}-list.patch
 }
 
 src_install () {
```

diff -u wmnd:
```
--- wmnd-0.4.17.ebuild  2018-07-22 19:43:12.973992678 +0200
+++ wmnd-0.4.17-r1.ebuild       2018-07-22 19:44:35.398868567 +0200
@@ -1,7 +1,7 @@
 # Copyright 1999-2018 Gentoo Foundation
 # Distributed under the terms of the GNU General Public License v2
 
-EAPI=4
+EAPI=7
 
 DESCRIPTION="WindowMaker Network Devices (dockapp)"
 HOMEPAGE="https://www.thregr.org/~wavexx/software/wmnd/"
@@ -19,9 +19,3 @@
        snmp? ( >=net-analyzer/net-snmp-5.2.1 )"
 DEPEND="${RDEPEND}
        x11-base/xorg-proto"
-
-src_install() {
-       emake DESTDIR="${D}" install
-
-       dodoc README AUTHORS ChangeLog NEWS TODO
-}
```

### CODE REVIEW

File: x11-plugins/wmmenu/wmmenu-1.3-r2.ebuild [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+}
+
+src_compile() {
+	emake  CC="$(tc-getCC)"
+}
+
+src_install() {
+	dobin wmmenu
+	doman wmmenu.1
+	einstalldocs
```
Comment (hydrapolic) [inline x11-plugins/wmmenu/wmmenu-1.3-r2.ebuild]: Wouldn't default work here?

File: x11-plugins/wmmon+smp/wmmon+smp-1.0-r3.ebuild [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+	default
+	# Respect LDFLAGS, see bug #335047
+	sed \
+		-e 's/cc -o/${CC} ${LDFLAGS} -o/' \
+		-e 's/cc -c/${CC} ${CFLAGS} -c/' \
+		-i Makefile || die
+	tc-export CC
+
+	cd "${WORKDIR}"/wmmon.app || die
+	eapply "${FILESDIR}"/${P}-list.patch
```
Comment (hydrapolic) [inline x11-plugins/wmmon+smp/wmmon+smp-1.0-r3.ebuild]: Moving into PATCHES array and utilizing the default from above?

Comment (mm1ke) [reply x11-plugins/wmmon+smp/wmmon+smp-1.0-r3.ebuild]: @hydrapolic 
Thanks again for reviewing, i put the patch into ```PATCHES``` and moved ```default``` after the ```cd```

File: x11-plugins/wmmenu/wmmenu-1.3-r2.ebuild (hunk 1 above)
Comment (hydrapolic) [reply x11-plugins/wmmenu/wmmenu-1.3-r2.ebuild]: I see, it's just a plain Makefile with hardcoded paths. Seems like multiple variables would have to be rewritten, in that case it's better to leave it like this.

**Comment by voyageur** [comment]
Thanks merged, catching up on my backlog!

**Pull request closed without merging**